File: Modules/ServerTypes/PerformanceAppraisal/PerformanceAppraisalTemplateDepartmentForm.ts
import { IntegerEditor, PrefixedContext } from "@serenity-is/corelib";
import { initFormType } from "@serenity-is/corelib/q";

export interface PerformanceAppraisalTemplateDepartmentForm {
    TemplateId: IntegerEditor;
    DepartmentId: IntegerEditor;
}

export class PerformanceAppraisalTemplateDepartmentForm extends PrefixedContext {
    static formKey = 'PerformanceAppraisal.PerformanceAppraisalTemplateDepartment';
    private static init: boolean;

    constructor(prefix: string) {
        super(prefix);

        if (!PerformanceAppraisalTemplateDepartmentForm.init)  {
            PerformanceAppraisalTemplateDepartmentForm.init = true;

            var w0 = IntegerEditor;

            initFormType(PerformanceAppraisalTemplateDepartmentForm, [
                'TemplateId', w0,
                'DepartmentId', w0
            ]);
        }
    }
}

File: Modules/HumanResource/PerformanceAppraisal/PerformanceAppraisalTemplateDepartment/PerformanceAppraisalTemplateDepartmentForm.cs
using Serenity.ComponentModel;
using System;

namespace HRMSoftware.PerformanceAppraisal.Forms;

[FormScript("PerformanceAppraisal.PerformanceAppraisalTemplateDepartment")]
[BasedOnRow(typeof(PerformanceAppraisalTemplateDepartmentRow), CheckNames = true)]
public class PerformanceAppraisalTemplateDepartmentForm
{
    public int TemplateId { get; set; }
    public int DepartmentId { get; set; }
}